src/App.tsx

```diff
@@ -15,36 +15,104 @@
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = '',
+}
+
+const sortGoods = (
+  goods: string[],
+  sortType: SortType,
+  isReverse: boolean,
+): string[] => {
+  let sortedGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.Alphabet:
+      sortedGoods = sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.Length:
+      sortedGoods = sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...goods];
+  }
+
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [activeSort, setActiveSort] = useState<SortType>(SortType.Default);
+  const [isReverse, setIsReverse] = useState(false);
+
+  const goods: string[] = sortGoods(goodsFromServer, activeSort, isReverse);
+
+  const sortAlphabetically = (): void => {
+    setActiveSort(SortType.Alphabet);
+  };
+
+  const sortByLength = (): void => {
+    setActiveSort(SortType.Length);
+  };
+
+  const goodsReverse = (): void => {
+    setIsReverse(prev => !prev);
+  };
+
+  const goodsReset = (): void => {
+    setActiveSort(SortType.Default);
+    setIsReverse(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${activeSort === 'alphabet' ? '' : 'is-light'}`}
+          onClick={sortAlphabetically}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${activeSort === 'length' ? '' : 'is-light'}`}
```

Comment:
Consider using the `SortType.Length` enum value instead of the string 'length' for consistency and to avoid potential errors if the enum value changes.

Reply:
> Consider using the `SortType.Length` enum value instead of the string 'length' for consistency and to avoid potential errors if the enum value changes.

Fix in all places